jabgui/src/main/java/org/jabref/gui/consistency/ConsistencyCheckAction.java

```diff
@@ -45,22 +47,39 @@ public ConsistencyCheckAction(Supplier<LibraryTab> tabSupplier,
 
     @Override
     public void execute() {
-        BackgroundTask.wrap(() -> {
-            BibDatabaseContext databaseContext = stateManager.getActiveDatabase()
-                                                             .orElseThrow(() -> new NullPointerException("Database null"));
-            List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+        Task<BibliographyConsistencyCheck.Result> task = new Task<>() {
+            @Override
+            public BibliographyConsistencyCheck.Result call() throws Exception {
+
+                Optional<BibDatabaseContext> databaseContext = stateManager.getActiveDatabase();
+                if (databaseContext.isEmpty()) {
+                    throw new IllegalStateException((Localization.lang("No library present")));
+                }
+                List<BibEntry> entries = databaseContext.get().getEntries();
 
-            BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
-            return consistencyCheck.check(entries);
-        }).onSuccess(result -> {
-            if (result.entryTypeToResultMap().isEmpty()) {
+                BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
+                return consistencyCheck.check(entries, (count, total) ->
+                        UiTaskExecutor.runInJavaFXThread(() -> {
+                            updateProgress(count, total);
+                            updateMessage(Localization.lang("%0/%1 entry types", count + 1, total));
```

Comment:
Not sure if we should use `%0 of %1` here, but maybe, this short form looks better?